packages/loader/container-loader/src/loader.ts

```diff
@@ -70,34 +62,25 @@ export class RelativeLoader extends EventEmitter implements ILoader {
 
     public async resolve(request: IRequest): Promise<IContainer> {
         if (request.url.startsWith("/")) {
-            // If no headers are set that require a reload make use of the same object
-            const container = await this.containerDeferred.promise;
-            return container;
+            if (canUseCache(request)) {
+                return this.container;
+            } else {
+                const resolvedUrl = this.container.resolvedUrl;
+                ensureFluidResolvedUrl(resolvedUrl);
+                return this.loader.resolve(request, undefined, resolvedUrl);
```

Comment:
don't use the loader here, just create the container

Reply:
Done.